lib/puppet/network/http/factory.rb

```diff
@@ -38,6 +38,13 @@ def create_connection(site)
     http.read_timeout = Puppet[:http_read_timeout]
     http.open_timeout = Puppet[:http_connect_timeout]
 
+    if Puppet[:sourceaddress]
+      msg = 'Setting source address is unsupported by this version of Net::HTTP.'
+      raise Net::HTTPError.new(msg, 400) unless http.respond_to?(:local_host)
```

Comment:
How about raising an `ArgumentError` instead of HTTP 400?

Reply:
Update made.  Yes, that's a much better choice for the user instead of just indicating that the library doesn't support the option.